### ISSUE TITLE
[REVIEW]: The o80 C++ templated toolbox: Designing customized Python APIs for synchronizing realtime processes

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2752]
**Submitting author:** @vincentberenz (<a href="http://orcid.org/0000-0003-2226-9660">Vincent Berenz</a>)
**Repository:** <a href="https://github.com/intelligent-soft-robots/o80" target ="_blank">https://github.com/intelligent-soft-robots/o80</a>
**Version:** v1.0
**Editor:** @gkthiruvathukal
**Reviewers:** @traversaro, @vissarion
**Archive:** <a href="https://doi.org/10.5281/zenodo.5357876" target="_blank">10.5281/zenodo.5357876</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/828f23ed755fafd07add9e6154b76b65/status.svg)](https://joss.theoj.org/papers/828f23ed755fafd07add9e6154b76b65)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/828f23ed755fafd07add9e6154b76b65"><img src="https://joss.theoj.org/papers/828f23ed755fafd07add9e6154b76b65/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/828f23ed755fafd07add9e6154b76b65/status.svg)](https://joss.theoj.org/papers/828f23ed755fafd07add9e6154b76b65)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/intelligent-soft-robots/o80" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@vissarion & @traversaro, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @gkthiruvathukal know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/intelligent-soft-robots/o80">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@vincentberenz) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/intelligent-soft-robots/o80">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@vincentberenz) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @giodegas, @traversaro it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02752/joss.02752/10.21105.joss.02752.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02752/joss.02752/10.21105.joss.02752.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1109/LRA.2018.2795645 is OK
- 10.1109/iros.2012.6386109 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by traversaro** [comment]
Hi @vincentberenz , I see that you submitted the version o80 v1.0 for review, but I can't find it in the repo any reference to any v1.0 version, can you clarify this? Thanks!

**Comment by vincentberenz** [comment]
Hi @traversaro , thanks a lot for accepting to review the paper. My apologies, I missed that a tagged/version had to set for review. The master branch is meant to always be the latest stable version. If needed, I'd be happy to tag the latest commit as version 1.0.

**Comment by traversaro** [comment]
> 
> 
> Hi @traversaro , thanks a lot for accepting to review the paper. My apologies, I missed that a tagged/version had to set for review. The master branch is meant to always be the latest stable version. If needed, I'd be happy to tag the latest commit as version 1.0.

Besides the specific requirement of the JOSS review, it seems to be that at least the repos https://github.com/intelligent-soft-robots/o80 are https://github.com/intelligent-soft-robots/o80_example are relatively tightly coupled (one is the software and one is the canonical example on how to use it). Unless you ensure perpetual API stability (that I could not see in the docs) as a user if I start developing against a given version of o80 it would be useful to know what is the corresponding version of https://github.com/intelligent-soft-robots/o80_example . This information probably can be obtained looking at the commit dates, but having a clear version to couple the two could be convenient for the user.

**Comment by traversaro** [comment]
Related to installation instructions, I opened an issue in the repo: https://github.com/intelligent-soft-robots/o80/issues/2 .

**Comment by gkthiruvathukal** [comment]
@traversaro  I think we're ok here and that review can proceed, right?  I think everything being reviewed should be tagged v1.0. You raise good points about the coupled example. Even though that might not show up in the submission metadata, it should be tagged v1.0, too. I think it is safe to proceed with review nevertheless. At the end, based on all issues raised, the authors will probably make some changes and tag everything again at the end. Hope that helps!

**Comment by traversaro** [comment]
Thanks, that is clear to me. For sure the review can continue and is not blocked by this, I just wanted to clarify things.

**Comment by vincentberenz** [comment]
@traversaro @gkthiruvathukal I tagged both o80 and o80_example with version 1.0. Thanks for the comments on the alignment of versions between o80 and o80_example, as well as for the issue related to the installation procedure and the broken links. 
This is my first submission to JOSS: what is the recommended way ? Answering comments/issues as they are arise, in an interactive fashion, or wait for the review to be done before addressing the issues ?

**Comment by traversaro** [comment]
>  This is my first submission to JOSS: what is the recommended way ? Answering comments/issues as they are arise, in an interactive fashion, or wait for the review to be done before addressing the issues ?

It is the first review of JOSS as well. I don't know if there is any recommendation on that, but looking at the other reviews, it seems that an interactive review seems to be possible. Personally I think that interactive review would be desirable, for example because I wanted to test the example, and I already found problems in following the docs for installation, so for me it would be ideal if the installation docs could be fixed, so that I could then try the example following the new installation docs.

**Comment by vincentberenz** [comment]
> > This is my first submission to JOSS: what is the recommended way ? Answering comments/issues as they are arise, in an interactive fashion, or wait for the review to be done before addressing the issues ?
> 
> It is the first review of JOSS as well. I don't know if there is any recommendation on that, but looking at the other reviews, it seems that an interactive review seems to be possible. Personally I think that interactive review would be desirable, for example because I wanted to test the example, and I already found problems in following the docs for installation, so for me it would be ideal if the installation docs could be fixed, so that I could then try the example following the new installation docs.

Interactive review sounds also great to me. I'll do so, and I will fix the docs soon.

**Comment by gkthiruvathukal** [comment]
@giodegas and  @traversaro, can you provide a brief update on the status of your reviews?

**Comment by traversaro** [comment]
Hi @gkthiruvathukal, I provided some preliminary comments to @vincentberenz on the installation procedure, that @vincentberenz addressed, now I need to install the software using the new docs, and play a bit with the examples to go forward with the review.

**Comment by giodegas** [comment]
I apologize, but over the last months I could not find any free time to concentrate on this task, as I hoped.
So I am refusing to review this work considering the back-log of my primary job duties.

**Comment by gkthiruvathukal** [comment]
@giodegas I totally understand. These are tough times for all of us. I'll see if I can add someone else.

**Comment by gkthiruvathukal** [comment]
@vincentberenz I apologize for the inconvenience but could you take another look at the list of reviewers and see if there is anyone else we could add as a second reviewer? I may need you to suggest 2-3 possibilities.

**Comment by giodegas** [comment]
I can suggest @PatrizioM , one of our PhD students, but he is also quite busy at the moment. Ask him.

**Comment by vincentberenz** [comment]
@gkthiruvathukal  no pb, here a list: jcarpent , davidbeckingsale , sarats , marcoapintoo, gcdeshpande , vissarion , vijaysm

**Comment by gkthiruvathukal** [comment]
Apologies for the delay in getting a replacement reviewer. 

@davidbeckingsale or @vissarion Would you be willing to contribute a review for this JOSS submission?

**Comment by vissarion** [comment]
@gkthiruvathukal I am OK with reviewing this submission. Is 4-5 weeks ok for a deadline?

**Comment by gkthiruvathukal** [comment]
Thanks for such a quick response, @vissarion. I'm willing to work with your schedule. We're still in a COVID-19 world, and your timeframe sounds perfect to me. I'm going to add you now.

**Comment by gkthiruvathukal** [comment]
@whedon add @vissarion as reviewer

**Comment by whedon** [comment]
OK, @vissarion is now a reviewer

**Comment by gkthiruvathukal** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Re-invite a reviewer (if they can't update checklists)
@whedon re-invite @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Set the software version at the top of the issue e.g.
@whedon set v1.0.1 as version

# Open the review issue
@whedon start review

EDITORIAL TASKS

# All commands can be run on a non-default branch, to do this pass a custom 
# branch name by following the command with `from branch custom-branch-name`.
# For example:

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Remind an author or reviewer to return to a review after a
# certain period of time (supported units days and weeks)
@whedon remind @reviewer in 2 weeks

# Ask Whedon to do a dry run of accepting the paper and depositing with Crossref
@whedon accept

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

EiC TASKS

# Invite an editor to edit a submission (sending them an email)
@whedon invite @editor as editor

# Reject a paper
@whedon reject

# Withdraw a paper
@whedon withdraw

# Ask Whedon to actually accept the paper and deposit with Crossref
@whedon accept deposit=true

```

**Comment by gkthiruvathukal** [comment]
@whedon remove @giodegas as reviewer

**Comment by whedon** [comment]
OK, @giodegas is no longer a reviewer

**Comment by gkthiruvathukal** [comment]
@whedon list reviewers

**Comment by whedon** [comment]
Here's the current list of reviewers: https://bit.ly/joss-reviewers

**Comment by davidbeckingsale** [comment]
> Apologies for the delay in getting a replacement reviewer.
> 
> @davidbeckingsale or @vissarion Would you be willing to contribute a review for this JOSS submission?

I don't have the bandwidth right now. Thanks for asking.

**Comment by gkthiruvathukal** [comment]
@traversaro and @vissarion, happy new year! Can. you let me know where you are with respect to the review for this JOSS submission?

**Comment by traversaro** [comment]
> 
> 
> @traversaro and @vissarion, happy new year! Can. you let me know where you are with respect to the review for this JOSS submission?

Hi @gkthiruvathukal ! Thanks for the ping. I plan to resume the review this week.

**Comment by gkthiruvathukal** [comment]
@traversaro Much appreciated!!

**Comment by vissarion** [comment]
@gkthiruvathukal happy new year! I paused my review waiting for a feedback in https://github.com/intelligent-soft-robots/o80/issues/3 since it seems there are some works in progress on the package.

**Comment by vincentberenz** [comment]
@gkthiruvathukal thanks for pinging this !
@vissarion @traversaro  : indeed, I am working on the packaging. For the build system, we are moving from catkin (used by ROS) to colcon/ament (used by ROS2). In parallel, I am creating a ppa to make installation of the compiled software trivial. All this is taking much more time and effort than expected. And also not as smooth as it should, you may notice the tag for compilation, unit tests and doc building are all red ... Obviously, this make reviewing much harder than it should (but it can wait, if ok with you). My apologies, switch to colcon and setup of a ppa were not part of the plan yet when I requested the review. Note that on the other hand the source code is not expected to change (beyond possible bug fixes).

**Comment by gkthiruvathukal** [comment]
@vincentberenz So can you give us a sense how long you'll need to make these changes? I can nudge everyone in a couple of weeks.

**Comment by vincentberenz** [comment]
@gkthiruvathukal I'd say around 2 weeks. I will ping this thread once up and running

**Comment by gkthiruvathukal** [comment]
@vincentberenz Just a nudge to see how things are going with the review. We greatly appreciate your help to complete it.

**Comment by vincentberenz** [comment]
@gkthiruvathukal things are progressing, but taking more time than expected. My apologies for making this review difficult.

**Comment by gkthiruvathukal** [comment]
@vincentberenz No worries here. Thank you for responding!

**Comment by vincentberenz** [comment]
@gkthiruvathukal @traversaro @vissarion 

The software is ready for the review to resume. Apologies again for this unexpected (long !) delay. The continuous integration has been fully refactored and now generates binaries and source tar balls. Here is the new documentation page:

http://people.tuebingen.mpg.de/mpi-is-software/o80/docs/o80/index.html

the installation page has been updated:

http://people.tuebingen.mpg.de/mpi-is-software/o80/docs/o80/doc/02.installation.html

**Comment by gkthiruvathukal** [comment]
Thanks, @vincentberenz. @traversaro and @vissarion, please let me know when you have a chance to take a look at the revisions.

**Comment by traversaro** [comment]
Hi @vincentberenz, thanks for the feedback, I will look into that the next week.

**Comment by vissarion** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02752/joss.02752/10.21105.joss.02752.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02752/joss.02752/10.21105.joss.02752.pdf) :page_facing_up: :point_left:

**Comment by vissarion** [comment]
Hi, I did another review pass after the revision and left a few comments as issues in the target repository:

- https://github.com/intelligent-soft-robots/o80/issues/5
- https://github.com/intelligent-soft-robots/o80/issues/6
- https://github.com/intelligent-soft-robots/o80/issues/7

**Comment by vincentberenz** [comment]
@vissarion Thanks a lot for your time and the comments. I agreed with all of them, and did my best to address them all. Please see answers in the issues.

**Comment by vincentberenz** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02752/joss.02752/10.21105.joss.02752.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02752/joss.02752/10.21105.joss.02752.pdf) :page_facing_up: :point_left:

**Comment by gkthiruvathukal** [comment]
@vissarion  and @traversaro Can you give me a sense of whether @vincentberenz  responses are sufficient? It seems like we are getting closer to acceptance.

**Comment by vissarion** [comment]
@vincentberenz thanks for your replies.
@gkthiruvathukal I am OK from my side.

**Comment by traversaro** [comment]
I plan to check the @vincentberenz replies are ok during this week.

**Comment by vincentberenz** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #2752 with the following error: 
```
 /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon.rb:204:in `block in parse_authors': Author (Vincent Berenz^[corresponding author]) is missing affiliation (RuntimeError)
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon.rb:202:in `each'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon.rb:202:in `parse_authors'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon.rb:93:in `initialize'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon/processor.rb:38:in `new'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon/processor.rb:38:in `set_paper'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/bin/whedon:58:in `prepare'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/bin/whedon:131:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `<main>'

```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02752/joss.02752/10.21105.joss.02752.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02752/joss.02752/10.21105.joss.02752.pdf) :page_facing_up: :point_left:

**Comment by vincentberenz** [comment]
I made a minor change in the paper: I updated the affiliations so that they align with some new guidelines of our institute

**Comment by vincentberenz** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02752/joss.02752/10.21105.joss.02752.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02752/joss.02752/10.21105.joss.02752.pdf) :page_facing_up: :point_left:

**Comment by gkthiruvathukal** [comment]
@vissarion and @traversaro Can you give me a sense of whether @vincentberenz latest updates are sufficient? The earlier correspondence suggests we are good to go. I'd like to move to accept this paper by end of week if there are no additional issues to address.

**Comment by vincentberenz** [comment]
@gkthiruvathukal 

Thanks for pinging !
I still have this issue to address:

https://github.com/intelligent-soft-robots/o80/issues/9

**Comment by gkthiruvathukal** [comment]
Thanks, @vincentberenz. Please do follow up when complete so we can take the next steps!

**Comment by vincentberenz** [comment]
@gkthiruvathukal @vissarion @traversaro It seems to me all issues have been addressed

**Comment by traversaro** [comment]
> 
> 
> @gkthiruvathukal @vissarion @traversaro It seems to me all issues have been addressed

Thanks, I will check this during the next week!

**Comment by traversaro** [comment]
Thanks @vincentberenz for addressing my comments! @gkthiruvathukal The review is ok from my side.

**Comment by gkthiruvathukal** [comment]
@traversaro Thanks for making sure your comments have been addressed. I'll be moving toward acceptance. There are some final steps to complete. Stay tuned.

**Comment by gkthiruvathukal** [comment]
Hi @traversaro, please do the following:

Please do the following:

- [ ] Make a tagged release of your software, and list the version tag of the archived version here. 
- [ ] Archive the reviewed software in Zenodo 
- [ ] Check the Zenodo deposit has the correct metadata, this includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it); you may also add the authors' ORCID.
- [ ] List the Zenodo DOI of the archived version here.

**Comment by traversaro** [comment]
> Hi @traversaro, please do the following:

I guess the message was intended for @vincentberenz ?

**Comment by gkthiruvathukal** [comment]
@traversaro Yes! That was the intention, but my fingers mentioned you instead of @vincentberenz. Thanks! I corrected it above.

**Comment by vincentberenz** [comment]
@traversaro @vissarion thanks again for your time and your useful comments

@gkthiruvathukal 

A robotic group of the University of Aachen  just finished developing o80 drivers for the Franka robots (which are currently quite popular in academia). I would like to update the paper with a reference to their repo. They told me it would make public very soon. As soon as this happens, I would update the paper accordingly (minimal change, smthg like "o80 drivers for the Franka robots are also availalbe [reference]"), and then go through the todo list you provided.

**Comment by gkthiruvathukal** [comment]
@vincentberenz This is absolutely fine. Please signal us when you have made the updates and have completed the checklist. The change you are proposing seems routine, so we do not need to do a full re-review by any means. Just make sure the article PDF generates properly!

**Comment by vincentberenz** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02752/joss.02752/10.21105.joss.02752.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02752/joss.02752/10.21105.joss.02752.pdf) :page_facing_up: :point_left:

**Comment by vincentberenz** [comment]
@gkthiruvathukal 

The o80's drivers for the Franka robot have been made open source, and I updated the paper to refer to them

**Comment by danielskatz** [comment]
👋 @gkthiruvathukal - what's the next step here?

**Comment by gkthiruvathukal** [comment]
@vincentberenz  I had already recommended this for acceptance. So can you go back to my August 1st comment and make sure to provide the needed information so we can bring this submission to closure?

**Comment by gkthiruvathukal** [comment]
@vincentberenz, So there are no changes to the code based on your point about the o80 drivers, right?
Thanks for your prompt response.

**Comment by vincentberenz** [comment]
@gkthiruvathukal I confirm the only change is the update of the paper (referring to the franka panda drivers). None of the code has been changed.

**Comment by gkthiruvathukal** [comment]
@vincentberenz  So I can just use your v1.0 tag for the release, right?

**Comment by gkthiruvathukal** [comment]
@whedon set 10.5281/zenodo.5357876 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5357876" target="_blank">10.5281/zenodo.5357876</a> is the archive.

**Comment by gkthiruvathukal** [comment]
@whedon set v1.0 as version

**Comment by whedon** [comment]
OK. v1.0 is the version.

**Comment by vincentberenz** [comment]
> @vincentberenz So I can just use your v1.0 tag for the release, right?

yes

**Comment by gkthiruvathukal** [comment]
@whedon accept

**Comment by whedon** [comment]
To recommend a paper to be accepted use `@whedon recommend-accept`

**Comment by gkthiruvathukal** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2642

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2642, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
👋 @vincentberenz - I've suggested some minor changes in the paper in https://github.com/intelligent-soft-robots/o80/pull/11 - please merge these or let me know what you disagree with, then we can proceed to acceptance and publication.

**Comment by vincentberenz** [comment]
@danielskatz Thanks for the improvements ! I merged the branch into master.

**Comment by danielskatz** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1109/LRA.2018.2795645 is OK
- 10.1109/iros.2012.6386109 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2643

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2643, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1445364477604646915) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2644
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.02752](https://doi.org/10.21105/joss.02752)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by danielskatz** [comment]
Congratulations to @vincentberenz (Vincent Berenz) and co-authors!!

And thanks to @traversaro and @vissarion for reviewing, and @gkthiruvathukal for editing!
We couldn't do this without you

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02752/status.svg)](https://doi.org/10.21105/joss.02752)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02752">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02752/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02752/status.svg
   :target: https://doi.org/10.21105/joss.02752
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02752/status.svg)](https://doi.org/10.21105/joss.02752)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by vincentberenz** [comment]
@gkthiruvathukal @traversaro @vissarion thanks indeed !
